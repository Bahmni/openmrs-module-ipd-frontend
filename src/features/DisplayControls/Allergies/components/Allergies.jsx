import {
  DataTable,
  DataTableSkeleton,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "carbon-components-react";
import React, { useEffect, useState } from "react";
import { useFetchAllergiesIntolerance } from "../hooks/useFetchAllergiesIntolerance";
import PropTypes from "prop-types";
import "../styles/Allergies.scss";
import { FormattedMessage } from "react-intl";

const Allergies = (props) => {
  const { patientId } = props;

  const { allergiesData, isLoading } = useFetchAllergiesIntolerance(patientId);
  const [rows, setRows] = useState([]);
  const NoAllergenMessage = (
    <FormattedMessage
      id={"NO_ALLERGENS_MESSAGE"}
      defaultMessage={"No Allergen is captured for this patient yet."}
    />
  );

  useEffect(() => {
    if (allergiesData && allergiesData.entry) {
      const allergies = [];
      allergiesData.entry?.map((allergy) => {
        allergies.push({
          allergen: allergy.resource.code.coding[0].display,
          id: allergy.resource.id,
          severity: getSeverity(allergy.resource.criticality),
          reaction: getAllergyReactions(allergy.resource.reaction),
          comments: getComments(allergy.resource.note),
          sortWeight: getSortingWait(getSeverity(allergy.resource.criticality)),
        });
      });
      setRows(sortedRow(allergies));
    }
  }, [allergiesData]);

  const getSortingWait = (severity) => {
    if (severity === "Severe") return -1;
    if (severity === "Moderate") return 0;
    return 1;
  };

  const getSeverity = (criticality) => {
    if (criticality == "unable-to-assess") return "Moderate";
    else if (criticality == "high") return "Severe";
    else return "Mild";
  };

  const getComments = (notes) =>
    notes && notes.length > 0 ? notes[0].text : "";

  const getAllergyReactions = (reactions) => {
    let allergyReactions = "";
    if (reactions && reactions.length > 0) {
      reactions[0].manifestation.map((reaction) => {
        allergyReactions =
          allergyReactions != ""
            ? `${allergyReactions}, ${reaction.coding[0].display}`
            : `${reaction.coding[0].display}`;
      });
    }
    return allergyReactions;
  };

  const headers = [
    {
      key: "allergen",
      header: "Allergen",
    },
    {
      key: "severity",
      header: "Severity",
    },
    {
      key: "reaction",
      header: "Reaction",
    },
    {
      key: "comments",
      header: "Comments",
    },
  ];

  const sortedRow = (rows) => {
    const sortedRows = rows
      .sort((a, b) => {
        if (a === b) return 0;
        return a.allergen > b.allergen ? 1 : -1;
      })
      .sort((a, b) => a.sortWeight - b.sortWeight);
    return sortedRows;
  };

  if (isLoading)
    return (
      <DataTableSkeleton
        data-testid="datatable-skeleton"
        headers={headers}
        aria-label="sample table"
        zebra="true"
      />
    );

  return allergiesData.entry === undefined ? (
    <div className="no-allergen-message"> {NoAllergenMessage} </div>
  ) : (
    <DataTable
      rows={rows}
      headers={headers}
      useZebraStyles={true}
      data-testid="datatable"
    >
      {({ rows, headers, getTableProps, getHeaderProps, getRowProps }) => (
        <Table {...getTableProps()} useZebraStyles>
          <TableHead>
            <TableRow>
              {headers.map((header, index) => (
                <TableHeader
                  key={index + header.key}
                  {...getHeaderProps({ header })}
                  isSortable={header.key === "severity"}
                >
                  {header.header}
                </TableHeader>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map((row, index) => (
              <TableRow
                key={index + row.id}
                {...getRowProps({ row })}
                data-testid="table-body-row"
              >
                {row.cells.map((cell) => (
                  <TableCell
                    key={cell.id}
                    className={
                      cell.id.includes("severity") && cell.value == "Severe"
                        ? "high-severity-color"
                        : ""
                    }
                  >
                    {cell.value}
                  </TableCell>
                ))}
              </TableRow>
            ))}
          </TableBody>
        </Table>
      )}
    </DataTable>
  );
};

Allergies.propTypes = {
  patientId: PropTypes.string.isRequired,
};

export default Allergies;
