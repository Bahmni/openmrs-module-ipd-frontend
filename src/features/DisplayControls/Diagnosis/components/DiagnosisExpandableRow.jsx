import React from "react";
import PropTypes from "prop-types";
import "../styles/DiagnosisExpandableRow.scss";

const DiagnosisExpandableRow = (props) => {
  const { data } = props;
  console.log("data inside DiagnosisExpandableRow", data);
  return (
    <>
      <h6 className="notes-heading">Notes</h6>
      <div className="notes-content">
        {data.diagnosisNotes ? data.diagnosisNotes : "No notes available"}
      </div>
      {data.diagnosisNotes !== "" && (
        <div className="notes-provider-details">
          {data?.provider} | {data.diagnosisTime}
        </div>
      )}
    </>
  );
};

DiagnosisExpandableRow.propTypes = {
  data: PropTypes.object,
};
export default DiagnosisExpandableRow;
