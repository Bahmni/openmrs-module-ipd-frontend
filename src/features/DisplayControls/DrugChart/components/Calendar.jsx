import React from "react";
import PropTypes from "prop-types";
import CalendarRow from "./CalendarRow.jsx";
import "../styles/Calendar.scss";
export default function Calendar(props) {
  const { calendarData, currentShiftArray, selectedDate } = props;
  return (
    <table className="drug-chart-calendar">
      <tbody>
        {calendarData.map((rowData, index) => {
          return (
            <tr key={index}>
              <CalendarRow
                rowData={rowData}
                currentShiftArray={currentShiftArray}
                selectedDate={selectedDate}
              />
            </tr>
          );
        })}
      </tbody>
    </table>
  );
}

Calendar.propTypes = {
  calendarData: PropTypes.array.isRequired,
  currentShiftArray: PropTypes.array,
  selectedDate: PropTypes.instanceOf(Date),
};
