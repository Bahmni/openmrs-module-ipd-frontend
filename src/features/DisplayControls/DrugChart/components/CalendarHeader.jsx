import React from "react";
import PropTypes from "prop-types";
import "../styles/CalendarHeader.scss";
import data from "../../../../utils/config.json";

export default function CalendarHeader(props) {
  const { currentShiftArray } = props;
  const enable24hour = data.config.drugChart.enable24HourTime;

  return (
    <div className="calendar-header">
      <div style={{ display: "flex" }}>
        {currentShiftArray.map((hour) => {
          const transformedHour = enable24hour ? hour : hour % 12 || 12;
          const formattedHour = `${transformedHour.toLocaleString("en-US", {
            minimumIntegerDigits: 2,
          })}:00`;

          return (
            <div data-testid="hour" key={hour} className={"hour-header"}>
              {formattedHour}
            </div>
          );
        })}
      </div>
    </div>
  );
}
CalendarHeader.propTypes = {
  currentShiftArray: PropTypes.array,
};
