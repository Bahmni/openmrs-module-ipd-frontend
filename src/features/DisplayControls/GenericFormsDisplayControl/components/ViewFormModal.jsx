import React, { Fragment, useEffect, useState } from "react";
import PropTypes from "prop-types";
import { Loading, Modal, Tooltip } from "carbon-components-react";
import { Document20 } from "@carbon/icons-react";
import _ from "lodash";
import {
  fetchAllObservations,
  flattenObservations,
} from "../utils/ViewFormUtils";
import "../styles/ViewFormModal.scss";
import moment from "moment/moment";

export const ViewFormModal = (props) => {
  const { encounterUuid, callbacks, form, metadata } = props;
  const { formTemplate, formTranslations } = form;
  const { formName, encounterDateTime, provider } = metadata;
  const [observations, setObservations] = useState({});
  const [isLoading, setIsLoading] = useState(false);
  useEffect(() => {
    setIsLoading(true);
    fetchAllObservations(encounterUuid).then((response) => {
      const flattenedObs = flattenObservations(response.observations);
      const groupedObs = _.groupBy(flattenedObs, "concept.name");
      setObservations(groupedObs);
      setIsLoading(false);
    });
  }, []);

  const renderMembers = (form, level, obs) => {
    const heading = (
      <span className={"observation-name"}>
        {Array(8 * level)
          .fill("\u00A0")
          .join("")}
        <span className={`${form.hasGroupMembers && "bold-text"} `}>
          {formTranslations[form.translationKey] &&
            formTranslations[form.translationKey][0]}
        </span>
      </span>
    );
    return (
      <>
        {!form.hasGroupMembers && obs[form.conceptName] && (
          <div className={"observations-row"}>
            {heading}
            <span className={"observation-value"}>
              {obs[form.conceptName][0].comment && (
                <Tooltip
                  renderIcon={Document20}
                  autoOrientation={true}
                  className={"obs-note-icon"}
                >
                  {obs[form.conceptName][0].comment}
                </Tooltip>
              )}
              {obs[form.conceptName]
                .map((value) => {
                  return value.valueAsString;
                })
                .join(", ")}
            </span>
          </div>
        )}
        {form.hasGroupMembers && (
          <div className={"obs-group-heading"}>{heading}</div>
        )}
        {form.hasGroupMembers &&
          form.members.map((member, index) => {
            return (
              <span key={index}>{renderMembers(member, level + 1, obs)}</span>
            );
          })}
      </>
    );
  };

  return (
    <Modal
      open
      passiveModal
      onRequestClose={() => {
        callbacks.setModalValue(encounterUuid, false);
      }}
      className={"view-form-modal"}
    >
      <div className={"view-form-container"}>
        <div className={"view-form-header"}>
          {formName} | {moment(+encounterDateTime).format("DD-MM-YYYY HH:mm")} |{" "}
          {provider}
        </div>
        {isLoading ? (
          <Loading />
        ) : (
          formTemplate.map((form, index) => {
            return (
              <Fragment key={index}>
                {renderMembers(form, 0, observations)}
              </Fragment>
            );
          })
        )}
      </div>
    </Modal>
  );
};

ViewFormModal.propTypes = {
  encounterUuid: PropTypes.string.isRequired,
  callbacks: PropTypes.object,
  form: PropTypes.object.isRequired,
  metadata: PropTypes.object.isRequired,
};
