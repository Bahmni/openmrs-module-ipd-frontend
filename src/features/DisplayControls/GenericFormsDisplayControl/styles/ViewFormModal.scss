@import "src/styles/styles.scss";

.view-form-modal {
  .observations-row {
    padding: 16px;
    border-bottom: 1px solid #e0e0e0;
    display: flex;

    .observation-name {
      display: inline-block;
      min-width: 40%;
    }
  }

  .obs-group-heading {
    padding: 16px;
    border-bottom: 1px solid #e0e0e0;
  }

  .bx--modal-container {
    width: 50%;
  }

  .bx--modal-close {
    top: 0;
    right: 0;
  }

  .observation-value {
    display: flex;
    align-items: center;
  }

  .bx--tooltip__trigger svg {
    fill: #0f62fe;
  }

  .view-form-container {
    border: 2px solid #007d79;
    border-radius: 6px;
  }

  .view-form-header {
    padding: 16px;
    color: #fff;
    background-color: #007d79;
  }

  @media screen and (max-width: $tablet-breakpoint) {
    .bx--modal-container {
      width: 65%;
    }
    .observation-name {
      min-width: 50%;
    }
  }
}
