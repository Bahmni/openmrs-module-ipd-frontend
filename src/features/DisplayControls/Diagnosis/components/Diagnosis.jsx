import React, { useEffect } from "react";
import { DataTableSkeleton } from "carbon-components-react";
import { useState } from "react";
import { FormattedMessage } from "react-intl";
import { getPatientDiagnosis, diagnosisHeaders } from "../utils/DiagnosisUtils";
import PropTypes from "prop-types";
import "../styles/Diagnosis.scss";
import { formatDate } from "../../../../utils/DateTimeUtils";
import ExpandableDataTable from "../../../../components/ExpandableDataTable/ExpandableDataTable";
import DiagnosisExpandableRow from "./DiagnosisExpandableRow";
import NotesIcon from "../../../../icons/notes.svg";

const Diagnosis = (props) => {
  const { patientId } = props;
  const [diagnosis, setDiagnosis] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const [additionalData, setAdditionalData] = useState([]);

  const NoDiagnosisMessage = (
    <FormattedMessage
      id={"NO_DIAGNOSIS_MESSAGE"}
      defaultMessage={"No Diagnosis found for this patient"}
    />
  );

  const mapDiagnosisData = (diagnosisList) => {
    const mappedDiagnoses =
      diagnosisList === undefined || !Array.isArray(diagnosisList)
        ? []
        : diagnosisList.map((diagnosis) => {
            let status = diagnosis.diagnosisStatusConcept
              ? "Inactive"
              : "Active";
            let diagnosisDate = formatDate(
              diagnosis.diagnosisDateTime,
              "DD/MM/YYYY"
            );
            let diagnosisTime = formatDate(
              diagnosis.diagnosisDateTime,
              "HH:mm"
            );
            let diagnosisId = diagnosis.codedAnswer
              ? diagnosis.codedAnswer.uuid
              : diagnosis.freeTextAnswer + diagnosis.diagnosisDateTime;
            let diagnosisNotes = diagnosis.comments ? diagnosis.comments : "";
            let diagnosisName = getDrugName(diagnosis);
            return {
              id: diagnosisId,
              diagnosis: diagnosisName,
              order: diagnosis.order,
              certainty: diagnosis.certainty,
              status: status,
              diagnosedBy: diagnosis.providers[0].name,
              diagnosisDate: diagnosisDate,
              additionalData: {
                diagnosisTime: diagnosisTime,
                diagnosisNotes: diagnosisNotes,
              },
            };
          });

    setDiagnosis(mappedDiagnoses);
    const additionalMappedData = mappedDiagnoses.map((diagnosis) => {
      return {
        id: diagnosis.id,
        diagnosisTime: diagnosis.additionalData.diagnosisTime,
        diagnosisNotes: diagnosis.additionalData.diagnosisNotes,
        provider: diagnosis.diagnosedBy,
      };
    });
    setAdditionalData(additionalMappedData);
    setIsLoading(false);
  };

  const getDrugName = (diagnosis) => {
    if (diagnosis.codedAnswer && diagnosis.comments !== "") {
      return (
        <div className="notes-icon-div">
          <NotesIcon className="notes-icon" />
          <span className="drug-name">{diagnosis.codedAnswer.name}</span>
        </div>
      );
    } else if (diagnosis.codedAnswer) {
      return diagnosis.codedAnswer.name;
    }
    return diagnosis.freeTextAnswer;
  };

  const getDiagnosis = async () => {
    const diagnosisList = await getPatientDiagnosis(patientId);
    mapDiagnosisData(diagnosisList);
  };

  useEffect(() => {
    getDiagnosis();
  }, []);

  return (
    <>
      {isLoading ? (
        <DataTableSkeleton data-testid="diagnosis-datatable-skeleton" />
      ) : diagnosis && diagnosis.length === 0 ? (
        <div className="no-dignosis-message">{NoDiagnosisMessage}</div>
      ) : (
        <ExpandableDataTable
          rows={diagnosis}
          headers={diagnosisHeaders}
          additionalData={additionalData}
          component={(additionalData) => {
            return <DiagnosisExpandableRow data={additionalData} />;
          }}
          useZebraStyles={true}
        />
      )}
    </>
  );
};

Diagnosis.propTypes = {
  patientId: PropTypes.string.isRequired,
};
export default Diagnosis;
