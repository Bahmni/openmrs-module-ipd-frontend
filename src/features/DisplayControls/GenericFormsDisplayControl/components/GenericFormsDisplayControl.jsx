import React, { useContext, useEffect, useState } from "react";
import PropTypes from "prop-types";
import { IPDContext } from "../../../../context/IPDContext";
import {
  fetchAllConceptsForForm,
  fetchFormTranslations,
} from "../utils/GenericFormsDisplayControlUtils";
import "../styles/GenericFormDisplayControl.scss";
import moment from "moment";
import {
  Link,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "carbon-components-react";
import { ViewFormModal } from "./ViewFormModal";

const GenericFormsDisplayControl = (props) => {
  const {
    config: { formName: formName },
  } = props;
  const ipdContext = useContext(IPDContext);
  const [formTemplate, setFormTemplate] = useState([]);
  const [formTranslations, setFormTranslations] = useState({});
  const [formEntries, setFormEntries] = useState([]);
  const [openViewFormModal, setOpenViewFormModal] = useState({});
  const headers = [
    {
      key: "Date",
      header: "Date",
    },
    {
      key: "Time",
      header: "Time",
    },
    {
      key: "Provider",
      header: "Provider",
    },
    {
      key: "",
      header: "",
    },
  ];
  const { allFormsSummary = [], allFormsFilledInCurrentVisit = [] } =
    ipdContext;
  const fetchFormConcepts = async (formUuid) => {
    const response = await fetchAllConceptsForForm(formUuid);
    console.log("response 1 ", response);
    if (response?.resources && response?.resources[0]?.value) {
      return JSON.parse(response?.resources[0]?.value);
    }
  };
  const restructureFormConcepts = (formConcepts) => {
    const form = [];
    formConcepts.map((formConcept) => {
      if (formConcept.type === "obsControl") {
        form.push({
          conceptName: formConcept.concept.name,
          translationKey: formConcept.label.translationKey,
          hasGroupMembers: false,
        });
      } else if (formConcept.type === "obsGroupControl") {
        form.push({
          conceptName: formConcept.concept.name,
          translationKey: formConcept.label.translationKey,
          hasGroupMembers: true,
          members: restructureFormConcepts(formConcept.controls),
        });
      }
    });
    return form;
  };

  useEffect(() => {
    const formUuid = allFormsSummary.find(
      (form) => form.name === formName
    )?.uuid;
    fetchFormConcepts(formUuid).then((response) => {
      console.log("response", response);
      const finalResponse = restructureFormConcepts(response.controls);
      setFormTemplate(finalResponse);
    });
    fetchFormTranslations(formName, formUuid).then((response) => {
      setFormTranslations(response.concepts);
    });
  }, []);

  useEffect(() => {
    getEntriesForSelectedForm();
  }, [allFormsFilledInCurrentVisit]);
  const setModalValue = (uuid, value) => {
    setOpenViewFormModal({ ...openViewFormModal, [uuid]: value });
  };

  const getEntriesForSelectedForm = () => {
    const forms = allFormsFilledInCurrentVisit.filter(
      (form) => form.formName === formName
    );
    forms.sort((a, b) => b.encounterDateTime - a.encounterDateTime);
    setFormEntries(forms);
  };
  console.log(
    "form",
    formTemplate,
    formTranslations,
    allFormsFilledInCurrentVisit,
    formEntries
  );
  return (
    <Table useZebraStyles className={"form-table-display-control"}>
      <TableHead>
        <TableRow>
          {headers.map((header, index) => (
            <TableHeader
              key={index + header.key}
              className={`${
                header.key === "Provider" && "provider-column-width"
              }`}
            >
              {header.header}
            </TableHeader>
          ))}
        </TableRow>
      </TableHead>
      <TableBody>
        {formEntries.map((formEntry, index) => {
          const { encounterDateTime, providers } = formEntry;
          const provider = providers[0].providerName;
          return (
            <TableRow key={index}>
              <TableCell>
                {moment(+encounterDateTime).format("DD-MM-YYYY")}
              </TableCell>
              <TableCell>
                {moment(+encounterDateTime).format("HH:mm")}
              </TableCell>
              <TableCell>{provider}</TableCell>
              <TableCell>
                <Link
                  onClick={() => {
                    setModalValue(formEntry.encounterUuid, true);
                  }}
                >
                  View
                </Link>
              </TableCell>
              {openViewFormModal[formEntry.encounterUuid] && (
                <ViewFormModal
                  encounterUuid={formEntry.encounterUuid}
                  callbacks={{ setModalValue }}
                  form={{ formTemplate, formTranslations }}
                  metadata={{
                    formName,
                    encounterDateTime: formEntry.encounterDateTime,
                    provider,
                  }}
                />
              )}
            </TableRow>
          );
        })}
      </TableBody>
    </Table>
  );
};

export default GenericFormsDisplayControl;

GenericFormsDisplayControl.propTypes = {
  patientId: PropTypes.string,
  config: PropTypes.object,
};
