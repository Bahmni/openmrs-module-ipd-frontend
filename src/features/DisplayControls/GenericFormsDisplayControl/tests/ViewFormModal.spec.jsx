import React from "react";
import { render, waitFor } from "@testing-library/react";
import { ViewFormModal } from "../components/ViewFormModal";
import {
  mockFormStructure,
  mockFormTranslations,
  mockObservations,
} from "./ViewFormModalMock";
import moment from "moment";
jest.mock("../utils/ViewFormUtils", () => {
  const originalModule = jest.requireActual("../utils/ViewFormUtils");
  return {
    ...originalModule,
    fetchAllObservations: jest.fn(() => {
      return Promise.resolve(mockObservations);
    }),
  };
});
describe("View Form Modal", () => {
  it("should match snapshot", async () => {
    const metadata = {
      formName: "Test Form",
      encounterDateTime: moment(1714145870000),
      provider: "Provider",
    };
    const { container, getByText } = render(
      <ViewFormModal
        form={{
          formTemplate: mockFormStructure,
          formTranslations: mockFormTranslations,
        }}
        metadata={metadata}
        encounterUuid={"b6cdc4e4-79e9-4f0e-a5de-a9e58aa3abad"}
      />
    );
    await waitFor(() => {
      expect(getByText("Image")).toBeTruthy();
    });
    expect(container).toMatchSnapshot();
  });
});
