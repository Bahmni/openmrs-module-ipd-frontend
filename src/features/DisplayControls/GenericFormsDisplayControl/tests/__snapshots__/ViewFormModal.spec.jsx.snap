// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Form Modal should match snapshot 1`] = `
<div>
  <div
    class="bx--modal is-visible view-form-modal"
    role="presentation"
  >
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
    <div
      aria-label=""
      aria-modal="true"
      class="bx--modal-container"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="bx--modal-header"
      >
        <button
          aria-label="close"
          class="bx--modal-close"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="bx--modal-close__icon"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
            />
          </svg>
        </button>
        
        <h3
          class="bx--modal-header__heading"
          id="bx--modal-header__heading--modal-1"
        />
      </div>
      <div
        aria-labelledby="bx--modal-header__heading--modal-1"
        class="bx--modal-content"
        id="bx--modal-body--modal-1"
      >
        <div
          class="view-form-container"
        >
          <div
            class="view-form-header"
          >
            Test Form
             | 
            26-04-2024 15:37
             | 
            Provider
          </div>
          <div
            class="observations-row"
          >
            <span
              class="observation-name"
            >
              
              <span
                class="false "
              >
                Image
              </span>
            </span>
            <span
              class="observation-value"
            >
              Image Value
            </span>
          </div>
          <div
            class="observations-row"
          >
            <span
              class="observation-name"
            >
              
              <span
                class="false "
              >
                Patient Video
              </span>
            </span>
            <span
              class="observation-value"
            >
              Patient Video Value
            </span>
          </div>
        </div>
      </div>
    </div>
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
  </div>
</div>
`;
